operations/model.go

```diff
@@ -155,9 +159,50 @@ func (s *ClientSettings) setupRestCommunicator(ctx context.Context, printMessage
 	if printMessages {
 		printUserMessages(ctx, c, !s.AutoUpgradeCLI)
 	}
+
+	if s.shouldGenerateJWT(ctx, c) {
+		grip.Info("We will attempt to generate a JWT token, to opt out of this, set 'do_not_run_kanopy_oidc' to true in your config file")
+		if s.JWT, err = RunKanopyOIDCLogin(); err != nil {
+			grip.Warningf("Failed to get JWT token: %s", err)
+			return c, err
+		}
+		c.SetJWT(s.JWT)
+	}
+
 	return c, nil
 }
 
+func (s *ClientSettings) shouldGenerateJWT(ctx context.Context, c client.Communicator) bool {
+	if s.DoNotRunKanopyOIDC {
+		return false
+	}
+
+	if s.APIKey == "" {
+		grip.Info("No API key found, attempting to use a JWT token.")
+		return true
+	}
+
+	flags, err := c.GetServiceFlags(ctx)
```

Comment:
Just from looking at the route, it seems like [the route it's requesting](https://github.com/evergreen-ci/evergreen/blob/e123dfbaf56c3cb64131e81164ea21882fe56714/rest/client/methods.go#L601) doesn't exist and a request to `evergreen.mongodb.com/rest/v2/admin` returns 404. It also looks like [all the admin routes except the one to get the banner are gated behind permission to view admin settings](https://github.com/evergreen-ci/evergreen/blob/e123dfbaf56c3cb64131e81164ea21882fe56714/rest/route/service.go#L115-L129), so I would expect this to always return an error for regular users, even if it requested the `/admin/service_flags` route. Do we need to make a special lower-permission route that users can access this flag?

Did we test the service flag toggling works in staging?